### DISCUSSION TITLE
pytest.warns and staticmethod

### DISCUSSION THREAD

**Discussion by sffjunkie** [pytest-dev/pytest#11179] [Q&A]
Does `pytest.warns` have a limitation when raising a warning from within a `staticmethod`?

If I wrap a `staticmethod` call in a `with pytest.warns` my test fails with a 'Failed: DID NOT WARN' but if I remove the `with` block then the warnings are printed to `stdout`

**Comment by RonnyPfannschmidt** [comment]
Static methods ought not to matter,
Please post some example code

**Comment by The-Compiler** [comment]
I agree with @RonnyPfannschmidt that your problem is somewhere else, but it's impossible to say more with so little information. This, for example, works just fine:

```python
import pytest
import warnings

class Warner:

    @staticmethod
    def warn():
        warnings.warn("Test", UserWarning)


def test_warner():
    warner = Warner()
    with pytest.warns(UserWarning, match="Test"):
        warner.warn()
```
